### DISCUSSION TITLE
Allow scss import per page basis

### DISCUSSION THREAD

**Discussion by gregg-cbs** [vercel/next.js#25293] [Ideas]
### Describe the feature you'd like to request

I would love to import page specific css files.
These css in js solutions have drawbacks which plain css doesnt have. 

Vue has page specific css so I know its possible. And you are allowing node_module css imports so you are kinda half way there.

Why is it that you do not want to allow importing of page specific css files?

### Describe the solution you'd like

if I have `home.js`, i would like to import a sass file related to home.js like `home.scss`.

The reason i want this is because css in js has drawbacks and limitations.
Like not being able to target something outside of the scope of the component or page. Same as css modules.

For example if I am on my home page and I want the header to have a blue background, I cannot do that easily in css in js or modules.

If i could import a css file I could literally say
```
.site-header{
  background: blue;
}
```

### Describe alternatives you've considered

I have looked into and tried:
Styled components, Styled system, atomize, rxcss and css modules. 

They all end up having catch cases and being a pain.
Nothing works as beautifully as normal css does.

Pleassseee, im begging you, i cannot take this js css any more. Its a pain. Always have to hack a way around a way.

**Comment by NateRadebaugh** [comment]
Do css modules not work for page components?
https://nextjs.org/docs/basic-features/built-in-css-support#adding-component-level-css

If you're looking for global styles at the page level, the only hack that's supported is if the styles are in `node_modules`:
* https://nextjs.org/docs/basic-features/built-in-css-support#import-styles-from-node_modules

Theoretically `css-loader` used by webpack should allow `:locale` vs `:global` but I haven't tried it myself and is likely discouraged:
* https://github.com/webpack-contrib/css-loader#scope

**Reply by gregg-cbs** [reply]
CSS modules work for pages and components.
But you cannot target anything outside of the page with css modules.
:global does not work.

With CSS modules you cant nest styles and do `composes` as well as other caveats I have run into.
 
If i can import and use a .module.scss stylesheet why can i not import a normal stylesheet?

-----

I cannot put my page css styles into node_modules, its an untracked folder.
And that is a hack that doesnt make sense because if i can import css from node_modules why cant i do it from any other folder?
Further more I would only be able to import css files and not sass files.

-----

Sass is simple. It works. Not allowing sass in pages has forced people to use and come up with solutions which are hacky and have holes in them. The internet is full of proof of this and people crying out about not having a solid solution.

**Comment by gregg-cbs** [comment]
I have figured a way to do this with Styled components.

This is what I settled on:
```javascript
const SignUp = () => {
  return (
    <PageStyle>
      <Head>
        <title>Get your business online | Create an account</title>
        <meta name="description" content="Time to operate. You've done the leg work, now let us get you online."/>
        <link rel="icon" href="/favicon.png"/>
      </Head>
      
      <Header/>

      <Container as="main">
        <SignUpContent />
      </Container>
    </PageStyle>
  )
}
  
  const PageStyle = styled.div`
    background-image: url("/svg/sitting-at-laptop.svg");
    background-repeat: no-repeat;
    background-position: -34vmax -3.5vmax;
    background-size: cover;
    flex: 1 auto;
  `
  
  export default SignUp;
```
